### DISCUSSION TITLE
Create a way that old packages can be transferred to new owners

### DISCUSSION THREAD

**Discussion by OwainWilliams** [umbraco/Umbraco.Packages#78] [Ideas] [answered]
After releasing my first package a conversation started on [Twitter](https://twitter.com/ScottishCoder/status/1298531688742625280) about how packages should be collaborated on rather than cloned. 
I cloned the package after a discussion with the previous owner but it did raise a couple of thoughts. 

It would be good if there was a way to transfer ownership of a package to a new developer if they wish to take over the maintenance and upkeep of that package. 
Obviously, you would need to have permission to take ownership of the package but this could be handled by maybe sending an email to the owner saying someone is interested in collaborating / taking ownership. 

I guess there are a couple of ways this could be handled but I thought it would be a good discussion to have and as Nik mentioned, it would be best to have that discussion here, rather than on twitter where it could be lost.

**Comment by jmayntzhusen** [comment]
I think it is a great idea, and will be watching the discussion and suggestions people may have on how best to achieve this!

Just a couple of notes:
- Ownership of a package means ownership of several different parts, the most common parts are ownership of the Github repo and of the Our Umbraco or NuGet package.
- It is currently possible to change the owner of a package on Our, but not in a documented or scalable way - you'd have to pretty much email an Our admin and have them do it from the backoffice. 
- Creating a feature where a package owner can first of all flag their package as looking for new maintainers to take over, and then being able to grant ownership would be really nice!

**Comment by abjerner** [comment]
The owner of a package can mark it as open for collaboration, which then lets the owner add additional contributors:

![image](https://user-images.githubusercontent.com/3634580/91351849-3de47500-e7e9-11ea-9ba6-38bc2dc6fb04.png)

This functionality already exists today. Ideally this functionality could be extended so a contributor can be promoted to a owner (also demoting the old owner to a collaborator).

Or should a package be able to have multiple owners? Could make sense if multiple developers inside a company are all developers of the package.

Also, if I was a newcomer, I would probably see contributors as outside developers helping with the package, and owners being part of the team behind the package. But to stay on topic, the naming etc. may be another discussion for another time.

**Comment by TimGeyssens** [comment]
1 thing that I think would be cool is to know who is using your package (agencies, devs), that could make it easier to spot if a package is in use a lot and if you don't feel up to maintaining it anymore another person can step in... Dunno if it is an idea to have an umbraco dev group on github that can take (shared) ownership of a repo if it is abandonment but other folks want to help...

**Comment by abjerner** [comment]
@TimGeyssens we do have https://github.com/umbraco-community/

It started from a similar idea to what you're describing. But I'm not sure how active it is at the moment. 

Personally I took over as maintainer of https://github.com/umbraco-community/Our.Umbraco.OpeningHours a few years ago, but I haven't really had the time to focus on it lately.

**Comment by TimGeyssens** [comment]
Ok nice, maybe a way of (if folks are up to it) adding an umb community account as co owner...with the soul goal of being able to step in when it is abandoned

**Comment by OwainWilliams** [comment]
I think the idea of being able to make a collaborator also an owner or even THE owner is a great idea. I think that would have solved the issue that I came across that I made a copy because the owner decided they don't have time to maintain the package just now or keep it updated. If I could have collabed and then became owner, this would have prevented a second package being on Our. 

Good to know that ownership can be transferred @jmayntzhusen by contacting an admin but as you say, this isn't really a permanent fix for the issue.

**Comment by TimGeyssens** [comment]
our is a platform HQ has control over, so the ownership at that stage isn't really the big issue... more the github abandoned project ...that needs a way of transfering (joined) ownership ... when the original creator drops out...

**Comment by OwainWilliams** [comment]
If it's in Github, could the option be that you fork the code at that point? Maybe even make a mention in the fork that it was originally part of projectX. Obviously this does require some timeframe to be in place e.g. if a project hasn't been updated for 1 year and you're not able to contact the owner, forking is an option?

**Comment by TimGeyssens** [comment]
Sure only thing that needs to be adressed if it is is linked to some build nuget deploy stuff... since having a package with a different name .... seems strange

**Comment by ronaldbarendse** [comment, marked as answer]
The Marketplace now uses NuGet as 'source-of-truth', so transferring ownership can be done by [managing the owners on the NuGet.org site](https://devblogs.microsoft.com/nuget/managing-package-owners/) and optionally also [transferring the repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository).
